concrete-core/src/specification/engines/glwe_secret_key_generation.rs

```diff
@@ -0,0 +1,43 @@
+use super::engine_error;
+use crate::specification::engines::AbstractEngine;
+use crate::specification::entities::GlweSecretKeyEntity;
+use concrete_commons::parameters::{GlweDimension, PolynomialSize};
+
+engine_error! {
+    GlweSecretKeyGenerationError for GlweSecretKeyGenerationEngine @
+    NullGlweDimension => "The secret key glwe dimension must be greater than zero.",
+    NullPolynomialSize => "The secret key polynomial size must be greater than zero.",
+    DegreeZeroPolynomial => "The secret key polynomial size must be greater than one. Otherwise\
+                             you should prefer the lwe scheme."
```

Comment:
I feel like the error name does not exactly match the error description, but I don't have alternatives

Reply:
By the way, I agree with @orelcosseron that we should use capital case for acronyms like LWE in the documentation and error messages.

Reply:
Good catch @tmontaigu ! I think `DegreeZeroPolynomial` is actually a poor name, because everywhere else we talk about polynomial size (degree + 1). What about changing it for `SizeOnePolynomial` ? Or something in this spirit ? Btw, @agnesLeroy, we expect the __polynomial size__ to be larger than __1__, to not have GLWE ciphertext with a single coefficient (what an lwe is...).

Reply:
Hum, I don't understand then: I thought `PolynomialSize` was the polynomial degree, `N`. It's not that?

Reply:
Ok, I replaced it by `SizeOnePolynomial` then !